src/main/java/org/openstreetmap/atlas/checks/validation/intersections/BuildingRoadIntersectionCheck.java

```diff
@@ -48,29 +59,41 @@
                         TunnelTag.YES)
                 || Validators.isOfType(edge, AreaTag.class, AreaTag.YES)
                 || YES_VALUE.equals(edge.tag(INDOOR_KEY))
-                || Validators.isOfType(edge, HighwayTag.class, HighwayTag.SERVICE)
+                || highwayServiceTag.test(edge)
                         && Validators.isOfType(edge, ServiceTag.class, ServiceTag.DRIVEWAY)
-                || edge.connectedNodes().stream().anyMatch(
-                        node -> Validators.isOfType(node, EntranceTag.class, EntranceTag.YES)
-                                || Validators.isOfType(node, AmenityTag.class,
-                                        AmenityTag.PARKING_ENTRANCE)));
+                || edge.connectedNodes().stream().anyMatch(node -> Validators.isOfType(node,
+                        EntranceTag.class, EntranceTag.YES)
+                        || Validators.isOfType(node, AmenityTag.class, AmenityTag.PARKING_ENTRANCE))
+                // Ignore edges with nodes containing Barrier tags
+                || edge.getAtlas()
+                        .nodesWithin(edge.bounds(),
+                                node -> Validators.hasValuesFor(node, BarrierTag.class))
+                        .iterator().hasNext());
     }
 
-    private static Predicate<Edge> intersectsCoreWayInvalidly(final Area building)
+    private Predicate<Edge> intersectsCoreWayInvalidly(final Area building)
     {
-        // An invalid intersection is determined by checking that the highway is a core way
-        return edge -> HighwayTag.isCoreWay(edge)
+        // An invalid intersection is determined by checking that its highway tag is car navigable
+        // or core way based on the configuration value
+        return edge -> (this.isCarNavigable ? HighwayTag.isCarNavigableHighway(edge)
+                : HighwayTag.isCoreWay(edge))
                 // And if the edge intersects the building polygon
                 && edge.asPolyLine().intersects(building.asPolygon())
                 // And ignore intersections where edge has highway=service and building has
                 // Amenity=fuel
-                && !(Validators.isOfType(edge, HighwayTag.class, HighwayTag.SERVICE)
+                && !(highwayServiceTag.test(edge)
                         && Validators.isOfType(building, AmenityTag.class, AmenityTag.FUEL))
+                // And ignore intersections where building has points within it with Amenity=fuel
+                && !edge.getAtlas().pointsWithin(building.asPolygon(),
```

Comment:
so you can do this processing outside of the lambda so you don't do this for every edge. 
```
if( ! building.getAtlas().pointsWithin ....... hasNext())
{
 //rest of logic
}else
{
return ture;
}
```

Reply:
You are right! As this condition is checked against buildings, I have moved this to the `validCheckForObject()`, which filters valid buildings for the Check.